### ISSUE TITLE
Possible wrong value client.supports_method call for semantic tokens

### ISSUE THREAD

**Issue by RayJameson** [AstroNvim/AstroNvim#2020]
### Checklist

- [X] Using a stable version of Neovim (i.e. not neovim nightly)
- [X] `:AstroUpdate`
- [X] Restarted AstroNvim

### Operating system/version

macOS 13.4.1

### Terminal/GUI

wezterm

### AstroNvim Health

AstroNvim ~
- AstroNvim Version: v3.25.0
- Neovim Version: v0.10.0-dev-552+g08db61b19-Homebrew
- WARNING Neovim nightly is not officially supported and may have breaking changes
- OK `git` is installed: Used for core functionality such as updater and plugin management
- OK `open` is installed: Used for `gx` mapping for opening files with system opener (Optional)
- OK `lazygit` is installed: Used for mappings to pull up git TUI (Optional)
- OK `node` is installed: Used for mappings to pull up node REPL (Optional)
- OK `gdu` is installed: Used for mappings to pull up disk usage analyzer (Optional)
- WARNING `btm` is not installed: Used for mappings to pull up system monitor (Optional)
- OK `python` is installed: Used for mappings to pull up python REPL (Optional)

### Describe the bug

I have warning messages when I open `python` file + conditional mapping `<leader>uY` is being set regardless
List of attached lsp servers:
- pyright
- ruff_lsp
- jedi_language_server
- null-ls

### Steps to Reproduce

1. Install listed lsp servers
2. Open any python file
3. Get warning

### Expected behavior

No warning and keymap `<leader>uY` shouldn't be set

### Screenshots

<img width="618" alt="image" src="https://github.com/AstroNvim/AstroNvim/assets/67468725/98e68368-d56a-486e-8551-7f1cc57b1bb1">

### Additional Context

I believe argument for method call `client.supports_method` is used wrong:
https://github.com/AstroNvim/AstroNvim/blob/4cd041082a8179f6d79b5f1addda1724c70d89e3/lua/astronvim/utils/lsp.lua#L340

It's supposed to be `textDocument/semanticTokens/full` and `textDocument/semanticTokens/full/delta` according to neovim source code:
https://github.com/neovim/neovim/blob/08db61b19b8fa07131fdcbea306893539963d4cb/runtime/lua/vim/lsp.lua#L61-L62

When I changed `textDocument/semanticTokens` to `textDocument/semanticTokens/full`, this warning stopped occurring, conditional mapping `<leader>uY` in that case also works fine

**Comment by Zeioth** [comment]
I think that is normal. I've opened a python file and either with 

    "textDocument/semanticTokens"

os

    "textDocument/semanticTokens/full" and "textDocument/semanticTokens/full/delta"

I get the warning on both. inlayHints also display a warning on typescript versions lower than 4.4. It would be nice having way to silence those errors, but I haven't succeeded so far. The parser shouldn't be throwing them in the first place, as there is nothing the user can do about them.

**Comment by RayJameson** [comment]
@Zeioth  can you try #2021 ?

**Comment by Zeioth** [comment]
@RayJameson idk the others but I still get in on my end with your patch

![screenshot_2023-06-22_12-28-30_509019920](https://github.com/AstroNvim/AstroNvim/assets/3357792/29daa17d-289f-43a2-992d-443e5a2a6b1a)
![screenshot_2023-06-22_12-28-46_059980409](https://github.com/AstroNvim/AstroNvim/assets/3357792/8c8ae5e8-bffb-481b-ad0f-7dbd7903c581)

**Comment by RayJameson** [comment]
@Zeioth this condition is wrong, it has to be inside brackets, because it is' `or` followed by `and`

make it like this
```lua
    (
      client.supports_method "textDocument/semanticTokens/full"
      or client.supports_method "textDocument/semanticTokens/full/delta"
    ) and vim.lsp.semantic_tokens
```

**Comment by Zeioth** [comment]
You are not forced to use parenthesis there because in boolean logic conditions are interpreted from left to right. Still, same result:
![screenshot_2023-06-22_12-43-36_876123025](https://github.com/AstroNvim/AstroNvim/assets/3357792/fb22458a-45d4-407c-9981-9be6832dca33)
![screenshot_2023-06-22_12-43-50_610494991](https://github.com/AstroNvim/AstroNvim/assets/3357792/230e91ad-16cf-42e2-94eb-bdc3550e0e2d)

**Comment by RayJameson** [comment]
Yea my bad, sometimes I forget about basic boolean logic

**Comment by mehalter** [comment]
@Zeioth this is not true, boolean logic is not done left to right it follows an order of operations, it's calculated NOT, then AND, then OR

**Comment by mehalter** [comment]
I can confirm the issue is fixed in the latest nightly @RayJameson

**Comment by Zeioth** [comment]
You are right @mehalter my bad there, checked it out. Still the patch don't do it on my end.

**Comment by RayJameson** [comment]
@Zeioth you are not changing anything in the `on_attach` file in your user config?

**Comment by RayJameson** [comment]
omfg dude... I'm blind as fuck, you made changes in the part where you set up mapping, not the call to activate semantic tokens

**Comment by Zeioth** [comment]
might be the case: I've cloned AstroNvim nightly and and it seems to be fixed there. I'm gonna research the cause on NormalNvim too.

**Comment by RayJameson** [comment]
You need to change it here as well
https://github.com/NormalNvim/NormalNvim/blob/e78bbb8ce6f7323231955867116050a0fe4439a9/lua/base/utils/lsp.lua#L470-L477

**Comment by Zeioth** [comment]
@RayJameson Thank you for noticing. I mixed my own implementation with astro's. It's fixed now on NormalNvim now too. Time for me to have some rest it seems.

**Comment by mehalter** [comment]
@Zeioth  it doesn't seem like it was your own implementation vs AstroNvim's. It seems that NormalNvim just had two semantic token blocks probably from the manual copying of our commits over to NormalNvim. From the commit previous to the fix:

- Original LSP Semantic Tokens block from AstroNvim that I wrote a while ago: https://github.com/NormalNvim/NormalNvim/blob/e78bbb8ce6f7323231955867116050a0fe4439a9/lua/base/utils/lsp.lua#L423-L431
- The updated block I wrote a day or two ago that was copied further down the file: https://github.com/NormalNvim/NormalNvim/blob/e78bbb8ce6f7323231955867116050a0fe4439a9/lua/base/utils/lsp.lua#LL470C1-L477C6

Also just for your awareness I did also notice that there are some inconsistencies in the commit messages/comments/release notes. Semantic Tokens is default enabled in stable Neovim v0.9 and is not new in Neovim v0.10.

**If you adopted a merge/rebase model on our code to preserve the original contributions of the commits this probably wouldn't have happened.**
